Framework/Sources/FloatingPanel.swift

```diff
@@ -596,27 +698,41 @@ class FloatingPanel: NSObject, UIGestureRecognizerDelegate {
             vc.delegate?.floatingPanelDidEndDragging(vc, withVelocity: velocity, targetPosition: targetPosition)
         }
 
-        if scrollView != nil, !stopScrollDeceleration,
+        guard let animator = self.animator else {
+            return
+        }
+
+        if
+            scrollView != nil,
+            !stopScrollDeceleration,
             surfaceView.frame.minY == layoutAdapter.topY,
-            targetPosition == layoutAdapter.topMostState {
+            targetPosition == layoutAdapter.topMostState,
+            animator.fractionComplete > 0.5
+        {
             self.state = targetPosition
             self.updateLayout(to: targetPosition)
             self.unlockScrollView()
-            return
```

Comment:
Was this line removed on purpose?

Reply:
Yes. Otherwise, the animation would not be completed when the `targetPosition` is `.full` but the user stopped dragging before the animation completed.